agenda/presentation/src/main/java/com/nibalk/tasky/agenda/presentation/home/HomeScreen.kt

```diff
@@ -26,78 +33,85 @@ import com.nibalk.tasky.agenda.presentation.model.AgendaItemActionType
-                Toast.makeText(
-                    context,
-                    event.error.asString(context),
-                    Toast.LENGTH_LONG
-                ).show()
+                Toast.makeText(context, event.error.asString(context), Toast.LENGTH_LONG).show()
             }
             is HomeEvent.FetchAgendaSuccess -> {
-                Toast.makeText(
-                    context,
-                    R.string.agenda_items_list_loaded,
-                    Toast.LENGTH_LONG
-                ).show()
+                Toast.makeText(context, R.string.agenda_items_list_loaded, Toast.LENGTH_SHORT).show()
             }
             is HomeEvent.LogoutError -> {
-                Toast.makeText(
-                    context,
-                    event.error.asString(context),
-                    Toast.LENGTH_LONG
-                ).show()
+                Toast.makeText(context, event.error.asString(context), Toast.LENGTH_LONG).show()
             }
             is HomeEvent.LogoutSuccess -> {
-                Toast.makeText(
-                    context,
-                    R.string.agenda_logout_successful,
-                    Toast.LENGTH_LONG
-                ).show()
+                Toast.makeText(context, R.string.agenda_logout_successful, Toast.LENGTH_SHORT).show()
             }
         }
     }
     HomeScreen(
         state = viewModel.state,
         onAction = { action ->
             when(action) {
-                is HomeAction.OnListItemOptionOpenClick -> onDetailClicked(false)
-                is HomeAction.OnListItemOptionEditClick -> onDetailClicked(true)
+                is HomeAction.OnAddAgendaOptionsClicked -> {
+                    onDetailClicked(true, action.agendaType, null)
+                }
+                is HomeAction.OnListItemOptionEditClicked -> {
+                    onDetailClicked(true, action.agendaType, action.agendaItem)
+                }
+                is HomeAction.OnListItemOptionOpenClicked -> {
+                    onDetailClicked(false, action.agendaType, action.agendaItem)
+                }
+                is HomeAction.OnListItemOptionDeleteClicked -> {
+                    Toast.makeText(
+                        context,
+                        "${getString(context, action.agendaType.menuNameResId)} " +
+                            "${getString(context, AgendaItemActionType.DELETE.menuNameResId)} " +
+                            "${getString(context, CoreR.string.toast_message_clicked)}!",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                }
                 else -> Unit
             }
             viewModel.onAction(action)
         }
     )
 }
 
+
 @SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
 @Composable
 fun HomeScreen(
     state: HomeState,
     onAction: (HomeAction) -> Unit
 ) {
-    val context = LocalContext.current
+    val now = LocalDateTime.now()
+    var showedNeedleOnce = false
```

Comment:
This should be a state or at least inside a remember block right? Right now it's a side-effect and will be reset on every recomposition.